docling_mcp/tools/manipulation.py

```diff
@@ -86,6 +87,99 @@ def get_overview_of_document_anchors(
     return DocumentAnchorOutput("\n".join(lines))
 
 
+@dataclass
+class TextSearchOutput:
+    """Output of the search_for_text_in_document_anchors tool."""
+
+    result: Annotated[
+        str,
+        Field(
+            description=(
+                "A string listing the result of searching for text in the document's "
+                "anchors. If matches were found, the result indicates what text matched "
+                "at which anchors, along with the number of occurrences. If no matches "
+                "were found, the result indicates that no matches were found."
+            )
+        ),
+    ]
+
+
+@mcp.tool()
+def search_for_text_in_document_anchors(
+    document_key: Annotated[
+        str,
+        Field(description="The unique identifier of the document in the local cache."),
+    ],
+    text: Annotated[
+        str,
+        Field(description="The string of text to search for in the document's anchors"),
+    ],
+) -> TextSearchOutput:
+    """Search for specific text and keywords within a document's anchors.
+
+    This tool takes a string of text to search for and returns a string of all
+    document anchors that contain the exact text. The search is case-insensitive.
+    If the exact text is not found, the tool will search for individual keywords
+    within the text, splitting it on non-alphanumeric characters. If keywords
+    are found, they are listed alongside their number of occurrences in parentheses.
+    """
+    if document_key not in local_document_cache:
+        doc_keys = ", ".join(local_document_cache.keys())
+        raise ValueError(
+            f"document-key: {document_key} is not found. Existing document-keys are: {doc_keys}"
+        )
+
+    doc = local_document_cache[document_key]
+    exact_matches = []
+    matches = []
+    keywords_set = {
+        word
+        for word in re.split(r"[~!@#$%^&*()_\-\[\]{}|\\:;\"'<>,.?/\s]+", text.lower())
+        if word
+    }
```

Comment:
We could make it simpler:

```suggestion
    keywords_set = set(re.findall(r'\b\w+\b', text.lower()))
```